# HG changeset patch
# User John Donoghue <john.donoghue@ieee.org>
# Date 1374166694 14400
#      Thu Jul 18 12:58:14 2013 -0400
# Node ID c2e43dd6eb708ff679ea9e1e608d7101667aa5d4
# Parent  84db9338c448ff777691c8214d38b166f035ee23
Dont display find dialog out of desktop bounds (bug #39518)

* libgui/src/m-editor/find-dialog.cc:
  (find_dialog::find_dialog): Check dialog bounds against desktop dize before settings dialog position.

diff --git a/libgui/src/m-editor/find-dialog.cc b/libgui/src/m-editor/find-dialog.cc
--- a/libgui/src/m-editor/find-dialog.cc
+++ b/libgui/src/m-editor/find-dialog.cc
@@ -153,17 +153,27 @@ find_dialog::find_dialog (QsciScintilla*
   main_layout->addWidget (_button_box, 0, 1);
   main_layout->addWidget (_extension, 1, 0);
   setLayout (main_layout);
 
   _extension->hide ();
   _find_next_button->setDefault (true);
   _find_result_available = false;
 
-  move (p->x() + p->frameGeometry ().width (), p->y());
+  // move to dialog to side of the parent if there is room on the desktop to do so.
+  QWidget * desktop = QApplication::desktop ();
+  int xp = p->x () + p->frameGeometry ().width ();
+  int yp=  p->y ();
+  if (desktop != 0 && sizeHint ().isValid ())
+    {
+      if (xp + sizeHint ().width () > desktop->width ())
+        xp = desktop->width () - sizeHint ().width ();
+    }
+
+  move (xp, yp);
 
 }
 
 // set text of "search from start" depending on backward search
 void
 find_dialog::handle_backward_search_changed (int backward)
 {
   if (backward)
